src/psyclone/domain/lfric/metadata_to_arguments_rules.py

```diff
@@ -0,0 +1,614 @@
+                    and func.function_space == function_space]:
+                cls._diff_basis(function_space)
+
+        # The boundary condition kernel (enforce_bc_kernel) is a
+        # special case.
+        if metadata.procedure_name and \
+                metadata.procedure_name.lower() == "enforce_bc_code":
+            cls._field_bcs_kernel()
+
+        # The operator boundary condition kernel
+        # (enforce_operator_bc_kernel) is a special case.
+        if metadata.procedure_name and \
+                metadata.procedure_name.lower() == "enforce_operator_bc_code":
+            cls._operator_bcs_kernel()
+
+        # Reference-element properties
+        if metadata.meta_ref_element:
+            cls._ref_element_properties(metadata.meta_ref_element)
+
+        # Mesh properties
+        if metadata.meta_mesh:
+            cls._mesh_properties(metadata.meta_mesh)
+
+        # Quadrature arguments are required if one or more basis or
+        # differential basis functions are used by the kernel and a
+        # quadrature shape is supplied.
+        if metadata.meta_funcs and metadata.shapes and \
+           [shape for shape in metadata.shapes if shape in
+                const.VALID_QUADRATURE_SHAPES]:
+            cls._quad_rule(metadata.shapes)
+
+    @staticmethod
+    def _field_meta_args_on_fs(arg_types, function_space, metadata):
+        '''Utility function to return any field (plus field vector, intergrid
+        or intergrid vector) meta_args in metadata that have the same
+        type as those specified in arg_types and are on the function
+        space specified in function_space.
+
+        :param arg_types: meta_arg classes indicating which meta_arg \
+            arguments to check.
+        :type arg_types: subclass of \
+            :py:class:`psyclone.domain.lfric.kernel.CommonMetaArgMetadata` or \
+            List[subclass of \
+            :py:class:`psyclone.domain.lfric.kernel.CommonMetaArgMetadata`]
+        :param str function_space: the specified function space.
+        :param metadata: the kernel metadata.
+        :type metadata: \
+            :py:class:`psyclone.domain.lfric.kernel.LFRicKernelMetadata`
+
+        :returns: a list of meta_args.
+        :type arg_types: List[subclass of \
+            :py:class:`psyclone.domain.lfric.kernel.CommonMetaArgMetadata`]
+
+        '''
+        return [arg for arg in metadata.meta_args_get(arg_types)
+                if arg.function_space == function_space]
+
+    @staticmethod
+    def _operator_meta_args_on_fs(arg_types, function_space, metadata):
+        '''Utility function to return any operator meta_args in metadata that
+        have the same type as those specified in arg_types and their
+        from or to function spaces are the same as the function space
+        specified in function_space.
+
+        :param arg_types: meta_arg classes indicating which meta_arg \
+            arguments to check.
+        :type arg_types: subclass of \
```

Comment:
I don't think we should have "subclass of" in the :type: fields.

Reply:
Removed throughout.